custom_model_runner/datarobot_drum/drum/typeschema_validation.py

```diff
@@ -0,0 +1,299 @@
+            elif n_columns != self.values[0]:
+                errors.append(
+                    "Num columns error, found {} but expected {} columns".format(
+                        n_columns, self.values[0]
+                    )
+                )
+        elif self.condition == "IN":
+            if not any([n_columns == value for value in self.values]):
+                errors.append(
+                    "Num columns error, found {} but expected number of columns to be in {}".format(
+                        n_columns, self.values
+                    )
+                )
+        elif self.condition == "NOT_EQUALS":
+            if len(self.values) > 1:
+                errors.append("Num columns error, only one value can be accepted for EQUALS")
+            elif n_columns == self.values[0]:
+                errors.append(
+                    "Num columns error, found {} columns, which is not supported".format(n_columns)
+                )
+        elif self.condition == "NOT_IN":
+            if any([n_columns == value for value in self.values]):
+                errors.append(
+                    "Num columns error, found {} columns, which is not supported".format(n_columns)
+                )
+        elif self.condition == "GREATER_THAN":
+            if len(self.values) > 1:
+                errors.append("Num columns error, only one value can be accepted for EQUALS")
+            elif n_columns <= self.values[0]:
+                errors.append(
+                    "Num columns error, found {} columns, but expected greater than {}".format(
+                        n_columns, self.values[0]
+                    )
+                )
+        elif self.condition == "LESS_THAN":
+            if len(self.values) > 1:
+                errors.append("Num columns error, only one value can be accepted for EQUALS")
+            elif n_columns >= self.values[0]:
+                errors.append(
+                    "Num columns error, found {} columns but expected less than {}".format(
+                        n_columns, self.values[0]
+                    )
+                )
+        else:
+            errors.append("Num columns error, unrecognized condition {}".format(self.condition))
+        return errors
+
+
+def get_type_schema_yaml_validator():
+    seq_validator = Seq(
+        Map(
+            {
+                "field": Enum(["data_types", "sparse", "number_of_columns"]),
+                "condition": Str(),
+                "value": Str() | Seq(Str()),
+            }
+        )
+    )
+    return Map(
+        {
+            Optional("input_requirements"): seq_validator,
+            Optional("output_requirements"): seq_validator,
+        }
+    )
+
+
+def revalidate_typeschema(type_schema):
+    """Revalidation of the elements in the lists for input and output requirements is required because there are
+    multiple valid sets of parameters and how the strictyaml validation works.  This checks that the provided values
```

Comment:
> Revalidation of the elements in the lists for input and output requirements is required because there are
    multiple valid sets of parameters and how the strictyaml validation works.  

This sentence is weird